docs/proposals/07-shoot-control-plane-migration.md

```diff
@@ -221,24 +221,44 @@ Horizontal Pod Autoscalers also need to be considered and removed.
+- If the timestamp has not been updated in the past `X` seconds, then the controllers will not reconcile the resources.
 
-Control plane components have to be shut down even when there is no access to the __Source Seed's__ API server. To be able to do that a daemonset is deployed in each Seed cluster. When the daemonset in the __Source Seed__ sees that it is nolonger the leader (by checking the DNS record) and there is no connection to the __Source Seed's__ API server, the daemonset will kill the Shoot's control plane pods by directly talking to the Kubelet's API server. If the __Source Seed's__ API Server comes back up, then gardenlet should take care of scaling down the deployments and statefulsets in the Shoot's control plane. This could be problematic if the gardenlet is in a crashloop backoff or takes too much time to do the scaling.
 
-As an alternative to the daemonset, a sidecar container can be added to each control plane component. The sidecar checks the DNS Entry to see if it is still the leader. If it is not, it shuts down the entire pod. This way we do not run the risk of deploymnets and statefulsets recreating the control plane pods after the seed's apiserver comes back up.
 
-The problems with using DNS as leader election is caching. Additionally, not all DNS servers respect TTL settings.
+**Alternative: DNS leader election:**
 
-**Using timestamps in the ETCD backup entries:** Once a Shoot is successfully created on the __Source Seed__ a timestamp is saved in the Cluster Resource and/or saved in the ETCD backup restore sidecar (either as an environment variable or additional configuration). The timestamps must not be modified afterwards and is used when the backup restore container writes data to the backup entry of the Shoot in the following way:
-- If there is no timestamp in the backup entry, the current timestamp is uploaded.
-- If there is a timestamp in the backup entry it is compared to the backup restore container's timestamp:
-    1. If it is the same, nothing is done
-    2. If it is older, it is replaced with the timestamp of the current backuprestore container
-    3. If it is newer - the current backuprestore container does not have ownership of the backup entry
+A DNS TXT entry with TTL=60s and value seed='__Source Seed__' is used. The record is created and maintained by the Gardener Controller Manager (by using the DNS Controller Manager and its DNSEntry resource). When a control plane migration is detected, the Gardener Controller Manager changes the value of the DNS Entry to seed='__Destination Seed__' and waits for 2*TTL + 1 = 121 seconds to ensure that the change is propagated to all controllers in the old seed. We rely on the fact that DNS is highly available (100% for AWS Route53) and that control plane components in the __Source Seed__ can see the changes.
```

Comment:
The same comments I made about leader election above holds here in the alternative too.